File: src/Scripting/Lua/pattern_sweep.lua
local files = {
  "command.lua", "interpreter.lua", "iterator.lua", "mediator.lua", "memento.lua", "observer.lua", "state.lua", "strategy.lua", "template_method.lua", "visitor.lua",
  "mvc.lua", "mvvm.lua", "microkernel.lua", "microservices.lua", "enterprise_adapter.lua", "enterprise_bridge.lua", "enterprise_facade.lua", "broker.lua", "message_bus.lua", "service_locator.lua",
  "active_object.lua", "monitor_object.lua", "half_sync_half_async.lua", "leader_followers.lua", "client_server.lua", "peer_to_peer.lua", "publish_subscribe.lua", "distributed_proxy.lua",
  "presentation_abstraction_control.lua", "model_view_presenter.lua", "document_view.lua", "active_record.lua", "data_mapper.lua", "unit_of_work.lua", "repository.lua",
  "dependency_injection.lua", "lazy_initialization.lua", "object_pool.lua", "null_object.lua"
}
local base = "src/Scripting/Lua/patterns/"
local passed = 0
for _, file in ipairs(files) do
  local ok, result = pcall(dofile, base .. file)
  assert(ok, file .. ": " .. tostring(result))
  assert(result == true, file .. ": expected true")
  passed = passed + 1
end
assert(passed == 39)
print(string.format("lua-pattern-sweep: %d/39 passed", passed))

File: src/Scripting/PHP/pattern_sweep.php
<?php declare(strict_types=1);
$patterns = [
'command','interpreter','iterator','mediator','memento','observer','state','strategy','template_method','visitor','mvc','mvvm','microkernel','microservices','enterprise_adapter','enterprise_bridge','enterprise_facade','broker','message_bus','service_locator','active_object','monitor_object','half_sync_half_async','leader_followers','client_server','peer_to_peer','publish_subscribe','distributed_proxy','presentation_abstraction_control','model_view_presenter','document_view','active_record','data_mapper','unit_of_work','repository','dependency_injection','lazy_initialization','object_pool','null_object'
];
$passed=0;
foreach ($patterns as $pattern) { require __DIR__."/patterns/$pattern.php"; $passed++; }
if ($passed !== 39) throw new RuntimeException("expected 39, got $passed");
echo "php-pattern-sweep: $passed/39 passed\n";
